contracts/minter-suite/Minters/MinterDAExpRefund/MinterDAExpRefundV0.sol

```diff
@@ -0,0 +1,786 @@
+            "Auction already marked as valid"
+        );
+        _projectConfig.selloutPrice = _newSelloutPrice;
+        emit SelloutPriceUpdated(_projectId, _newSelloutPrice);
+    }
+
+    /**
+     * @notice This represents an admin agreeing with the auction's sellout
+     * price (if it is a sellout auction) and marking the auction as valid.
+     * This is only callable by the core admin, only after the auction has
+     * sold out.
+     * This function is only required to be called in the case where an auction
+     * is a sellout auction, in which case it must be called before the artist
+     * or admin can withdraw funds. It is not required to be called in the case
+     * of a non-sellout auction, because in that case, the auction reached its
+     * base price, which is the lowest possible price, and therefore is
+     * considered valid and fair by default.
+     * This is intended to provide a mechanism for the admin to prevent an
+     * artist from withdrawing funds from an auction that has sold out, but did
+     * not sell out at a price that the admin deems to be fair. See the comment
+     * in the `adminEmergencyReduceSelloutPrice` function for examples.
+     * @param _projectId Project ID to mark auction as valid for.
+     */
+    function adminValidateSelloutPrice(uint256 _projectId)
+        external
+        onlyCoreAdminACL(this.adminValidateSelloutPrice.selector)
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        // may only be called after the auction has hit max invocations while
+        // minting on this minter.
+        require(_projectConfig.maxHasBeenInvoked, "Auction must be sellout");
+        // only callable when the sellout price is greater than auction base price
+        require(
+            _projectConfig.selloutPrice > projectConfig[_projectId].basePrice,
+            "Only sellout price greater than base price"
+        );
+        // update state
+        _projectConfig.auctionIsValid = true;
+        emit SelloutPriceValidated(_projectId, _projectConfig.selloutPrice);
+    }
+
+    /**
+     * @notice This withdraws project revenues for the artist and admin.
+     * This function is only callable by the artist or admin, and only after
+     * one of the following is true:
+     * - The auction has sold out above base price, and the admin has validated
+     *   the sellout price.
+     * - The auction has reached base price, and therefore is considered valid
+     *   by default.
+     */
+    function withdrawArtistAndAdminRevenues(uint256 _projectId)
+        external
+        onlyCoreAdminACLOrArtist(
+            _projectId,
+            this.withdrawArtistAndAdminRevenues.selector
+        )
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        // get the current net price of the auction
+        uint256 _price = _getPrice(_projectId);
+        // if the price is not base price, the auction is only valid if the
```

Comment:
May be worth noting that this is a reasonable assumption given that price is monotonically decreasing.

Reply:
updated!